### PR TITLE
[TD-5513]<feature>:cumulative sum/moving average/sample function

### PR DESCRIPTION
**Pull request by shenglian-zhou**

### CODE REVIEW

File: tests/script/general/compute/testSuite.sim [hunk 1]
```diff
@@ -3,6 +3,10 @@ run general/compute/bottom.sim
 run general/compute/count.sim
 run general/compute/diff.sim
 run general/compute/diff2.sim
+run general/compute/csum.sim
```
Comment (dapan1121) [inline tests/script/general/compute/testSuite.sim:6]: 为什么没有sample.sim

File: tests/script/regressionSuite.sim [hunk 2]
```diff
@@ -21,6 +21,10 @@ run general/compute/bottom.sim
 run general/compute/count.sim
 run general/compute/diff.sim
 run general/compute/diff2.sim
+run general/compute/mavg.sim
```
Comment (dapan1121) [inline tests/script/regressionSuite.sim:24]: sample.sim?

File: tests/script/general/compute/testSuite.sim (hunk 1 above)
Comment (shenglian-zhou) [reply tests/script/general/compute/testSuite.sim:6]: it shall be added

File: tests/script/regressionSuite.sim (hunk 2 above)
Comment (shenglian-zhou) [reply tests/script/regressionSuite.sim:24]: it shall be added.

File: src/client/src/tscSQLParser.c [hunk 3]
```diff
@@ -5554,7 +5586,8 @@ int32_t validateFillNode(SSqlCmd* pCmd, SQueryInfo* pQueryInfo, SSqlNode* pSqlNo
   size_t numOfExprs = tscNumOfExprs(pQueryInfo);
   for(int32_t i = 0; i < numOfExprs; ++i) {
     SExprInfo* pExpr = tscExprGet(pQueryInfo, i);
-    if (pExpr->base.functionId == TSDB_FUNC_TOP || pExpr->base.functionId == TSDB_FUNC_BOTTOM) {
+    if (pExpr->base.functionId == TSDB_FUNC_TOP || pExpr->base.functionId == TSDB_FUNC_BOTTOM
+        || pExpr->base.functionId == TSDB_FUNC_SAMPLE) {
       return invalidOperationMsg(tscGetErrorMsgPayload(pCmd), msg3);
```
Comment (dapan1121) [inline src/client/src/tscSQLParser.c:5591]: msg3里没有sample

File: src/query/src/qAggMain.c [hunk 4]
```diff
@@ -4483,6 +4539,310 @@ static void round_function(SQLFunctionCtx *pCtx) {
...
+
+      *pTimestamp = (tsList != NULL) ? tsList[i] : 0;
+      SET_DOUBLE_VAL(pOutput, mavgInfo->sum / mavgInfo->numPointsK)
+
+      ++notNullElems;
+      pOutput += pCtx->outputBytes;
+      pTimestamp++;
+    }
+
+    ++mavgInfo->pos;
```
Comment (dapan1121) [inline src/query/src/qAggMain.c:4664]: 当表中数据量足够大时pos会溢出吧

**Pull request merged**